pkg/util/https/server.go

```diff
@@ -43,23 +53,63 @@ type Server struct {
 // NewServer returns a Server instance.
 func NewServer(certFile, keyFile string) *Server {
 	mux := http.NewServeMux()
-	tls := &http.Server{
-		Addr:    ":8081",
-		Handler: mux,
-	}
 
 	wh := &Server{
 		Mux:      mux,
-		tls:      tls,
 		certFile: certFile,
 		keyFile:  keyFile,
 	}
+	wh.setupServer()
+
 	wh.Mux.HandleFunc("/", wh.defaultHandler)
 	wh.logger = runtime.NewLoggerWithType(wh)
 
 	return wh
 }
 
+func (s *Server) setupServer() {
+	s.tls = &http.Server{
+		Addr:    ":8081",
+		Handler: s.Mux,
+		TLSConfig: &cryptotls.Config{
+			GetCertificate: s.getCertificate,
+		},
+	}
+
+	// Start a goroutine to watch for certificate changes
+	go s.watchForCertificateChanges()
```

Comment:
This doesn't need to be a gorountine, just `s.watchForCertificateChanges()` should be fine

Reply:
Sure, I made `WatchForCertificateChanges(`) function public so that it should be accessible from extensions/controller.

Reply:
Sure, I made `WatchForCertificateChanges()` function public so that it should be accessible from extensions/controller.